test/ered_tls_SUITE.erl

```diff
@@ -0,0 +1,192 @@
+-module(ered_tls_SUITE).
+
+-include("ered_test_utils.hrl").
+
+-compile([export_all, nowarn_export_all]).
+
+all() ->
+    [t_command,
+     {group, require_faketime}].
+
+groups() ->
+    %% Tests that require 'faketime' to manipulate the system time.
+    [{require_faketime, [sequence],
+      [t_expired_cert_tls_1_2,
+       t_expired_cert_tls_1_3]}].
+
+-define(PORTS, [31001, 31002, 31003, 31004, 31005, 31006]).
+
+-define(DEFAULT_SERVER_DOCKER_IMAGE, "valkey/valkey:8.0.1").
+
+-define(TLS_OPTS, [{cacertfile, "tls/ca.crt"},
+                   {certfile,   "tls/client.crt"},
+                   {keyfile,    "tls/client.key"},
+                   {verify,     verify_peer},
+                   {server_name_indication, "Server"}]).
+
+-define(CLIENT_OPTS, [{connection_opts, [{tls_options, ?TLS_OPTS}]}]).
+
+init_per_suite(_Config) ->
+    stop_containers(), % just in case there is junk from previous runs
+    generate_tls_certs(),
+    start_containers(),
+    create_cluster(),
+    ered_test_utils:wait_for_consistent_cluster(?PORTS, ?CLIENT_OPTS),
+    [].
+
+end_per_suite(_Config) ->
+    stop_containers().
+
+init_per_group(require_faketime, _Config) ->
+    case os:find_executable("faketime") of
+        false ->
+            {skip, "Executable faketime not found"};
+        _ ->
+            ok
+    end.
+
+end_per_group(require_faketime, _Config) ->
+    ok.
+
+init_per_testcase(_Testcase, Config) ->
+    %% Make sure we have a valid client cert.
+    generate_client_cert(),
+
+    %% Quick check that cluster is OK; otherwise restart everything.
+    case catch ered_test_utils:check_consistent_cluster(?PORTS, ?CLIENT_OPTS) of
+        ok ->
+            [];
+        _ ->
+            ct:pal("Re-initialize the cluster"),
+            init_per_suite(Config)
+    end.
+
+generate_tls_certs() ->
+    filelib:ensure_path("tls/"),
```

Comment:
When using rebar3, as we do in CI and probably most users, the current directory path is set by the common test runner.
The tls files are placed like:
`<ered repo>/_build/test/logs/ct_run.nonode@nohost.2024-11-18_15.42.48/tls/client.crt`

These files are not overwritten since rebar3 creates a new directory for each run, which might be good when reproducing an issue.